[Rohit Rajesh Bhat] iP

rohit0718 opened the pull request:
# DukeMaxPlus 😎 

> "why do something when you can just dont" - Einstein ([source](https://www.youtube.com/watch?v=dQw4w9WgXcQ))

DukeMaxPlus is barely usable. It is:
- a GUI app
- ~~Easy~~ **NOT THAT** _EASY_ to use

Steps to use:
1. download from [here](https://github.com/rohit0718/ip)
2. install and execute the gradle `run` task
3. done!

Cool code snippets (im a `regex` noob):
```java
if (!s.matches("[T] [|] [10] [|] [\\w\\s]+$")
        && !s.matches("[DE] [|] [10] [|] [\\w\\s]+ [|] [0-9]{2}/[0-9]{2}/[0-9]{4} [0-9]{4}")) {
    throw new IllegalArgumentException();
}
```

## TODO
- [x] gui
- [ ] nothing much

Commit:
Add Gradle support

rohit0718 committed:
added basic greet, echo, and exit functionality

rohit0718 committed:
added 'add' and 'list' methods

rohit0718 committed:
added Task class

rohit0718 committed:
added complete task method

rohit0718 committed:
added event task and todo classes

rohit0718 committed:
added 'addTodo' method

rohit0718 committed:
added 'addDeadline' method

rohit0718 committed:
added 'addEvent' method

rohit0718 committed:
added tests

rohit0718 committed:
added error handling for no tasks

rohit0718 committed:
added delete method

rohit0718 committed:
add functionality to read tasks from file

rohit0718 committed:
add functionality to save to file

rohit0718 committed:
change events to store dates as date objects instead of strings

rohit0718 committed:
Merge branch 'branch-Level-8'

rohit0718 committed:
fix file io to work with datetime objects

rohit0718 committed:
update tests with new date format

rohit0718 committed:
add storage class

rohit0718 committed:
add parser and ui classes

rohit0718 committed:
add duke package

rohit0718 committed:
add tests for todo and utility classes

rohit0718 committed:
add javadocs for non obvious public methods

rohit0718 committed:
follow convention for double indents for wrapped lines

rohit0718 committed:
add find method

rohit0718 committed:
Merge branch 'branch-A-JavaDoc'

rohit0718 committed:
Merge branch 'branch-A-CodingStandard'

rohit0718 committed:
Merge branch 'branch-Level-9'

rohit0718 committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-Gradle

rohit0718 committed:
update config for gradle tests

rohit0718 committed:
Merge branch 'branch-A-Gradle'

rohit0718 committed:
fix checkstyle errors for main and test files

rohit0718 committed:
Merge branch 'branch-A-CheckStyle'

rohit0718 committed:
add initial javafx config

rohit0718 committed:
add initial gui logic for echoing user commands

rohit0718 committed:
use fxml for gui implementation

rohit0718 committed:
update config to work with existing duke app

rohit0718 committed:
ui update to make images circular

rohit0718 committed:
fix checkstyle errors

rohit0718 committed:
Merge branch 'branch-Level-10'

ryantianj commented on src/main/java/Duke/Deadline.java:
```diff
@@ -0,0 +1,28 @@
+public class Deadline extends Task {
+    private final LocalDateTime dueDate;
+
+    Deadline(String name, String dueDate) throws DateTimeParseException {
```
Perhaps including access modifiers here would have made the code easier to read.

rohit0718 replied:
the default access modifier refers to a package private constructor here, which is intended!

ryantianj commented on src/main/java/Duke/Parser.java:
```diff
@@ -0,0 +1,161 @@
+        "deadline", this::addDeadline,
+        "event", this::addEvent,
+        "delete", this::deleteTask,
+        "find", this::findTasks
```
I like how you made use of Mapping here to execute commands!

ryantianj commented on src/main/java/Duke/Parser.java:
```diff
@@ -0,0 +1,161 @@
+    /**
+     * Executes a command represented by input string and updates provided tasklist.
+     *
+     * @param input command to execute
```
All Javadoc sentences should end with a fullstop, including the @param, @return statements etc.

ryantianj commented on src/main/java/Duke/Parser.java:
```diff
@@ -0,0 +1,161 @@
+
+    private String completeTask(String args, TaskList tasks) {
+        try {
+            int idx = Integer.parseInt(args);
```
Perhaps naming in full (index) would have been more clear instead!

ryantianj commented on src/main/java/Duke/Parser.java:
```diff
@@ -0,0 +1,161 @@
+    private static final String ERR_FIND_FORMAT = "Please provide a query! Usage: find <query>";
+    private static final String ERR_MAX_TASKS = "Sorry! You have reached maximum Task capacity.";
+    private static final String ERR_NO_MATCHES = "No tasks match given query.";
+    private static final int MAX_TASKS = 100;
```
Great job following final naming conventions!

ryantianj reviewed (commented):
Overall, coding standards were great (No major issues). Just some minor violations such as lack of punctuation in Javadocs, short forms being used, and perhaps some variable names could have been better. Could not find any fault with the coding format(indentation and the likes) :)

rohit0718 committed:
fix styling violations

rohit0718 commented:
@ryantianj noted on the comments and have made the remedies. thank you for the feedback!

justintanyf commented on src/main/java/Duke/Utility.java:
```diff
@@ -0,0 +1,45 @@
+import java.time.format.DateTimeParseException;
+
+
+public class Utility {
```
I liked the good coding conventions!

justintanyf commented on src/main/java/ui/DialogBox.java:
```diff
@@ -0,0 +1,71 @@
+    }
+
+    public static DialogBox getUserDialog(String text, Image img) {
+        DialogBox db = new DialogBox(text, img);
```
I think coding conventions mentioned that we should not use shortform for variable names.

rohit0718 replied:
thanks for pointing this out!

justintanyf reviewed (commented):
Overall an excellent job, I'm impressed with the work put in!

rohit0718 committed:
Modify build.gradle with jar dependencies

Current build.gradle setup does not bundle required dependencies when
jar file is created as cross-platform dependencies are absent.

Let's add the needed dependencies to the build.gradle file to allow the
jar file to be cross-platform compatible without requiring users to
download javafx dependencies.

rohit0718 committed:
Add assertion checks for TaskList

Currently TaskList does not have checks in place for out of bounds
access in the get, remove, and markComplete methods.

Let us add assertion checks to ensure that all access to the ArrayList
in TaskList are within bounds.

rohit0718 committed:
Rename variable to better represent underlying obj

Some variable names in DialogBox.java are ambiguous, leading to code
that is less readable.

Let us rename the variables to be better represent the object that they
are referring to in DialogBox.java

rohit0718 committed:
Change save file to data nested folder

Current setup clutters the root directory as it saves the savefile in
the root.

Let us change the save file to be written to and read from a nested data
folder instead.

rohit0718 committed:
Merge pull request #1 from rohit0718/branch-A-Assertions

Add assertion checks for TaskList

rohit0718 committed:
Merge branch 'master' of github.com:rohit0718/ip

rohit0718 committed:
Merge branch 'master' into branch-A-CodeQuality

rohit0718 committed:
Merge pull request #2 from rohit0718/branch-A-CodeQuality

Minor code quality fixes

rohit0718 committed:
Merge branch 'master' of github.com:rohit0718/ip

rohit0718 committed:
add Github ci push and pr workflow

rohit0718 committed:
add better search functionality

rohit0718 committed:
Merge branch 'branch-C-BetterSearch'

rohit0718 committed:
enable assertions for gradle run task

rohit0718 committed:
change images to transparent bg

rohit0718 committed:
update dialogbox for variable height

rohit0718 committed:
add readme docs
